src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/kv_cache_sdpa.cpp

```diff
@@ -0,0 +1,340 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/common_utils.hpp"
+#include "common_test_utils/ov_tensor_utils.hpp"
+#include "common_test_utils/ov_test_utils.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+#include "openvino/core/partial_shape.hpp"
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/scaled_dot_product_attention.hpp"
+#include "openvino/runtime/intel_gpu/properties.hpp"
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "shared_test_classes/base/utils/compare_results.hpp"
+#include "subgraphs_builders.hpp"
+#include "transformations/op_conversions/scaled_dot_product_attention_decomposition.hpp"
+
+namespace {
+using ov::test::InputShape;
+
+struct Params {
+    bool with_rearrange;
+    bool with_mask;
+    bool with_scale;
+    bool causal;
+    size_t batch;
+    ov::element::Type model_element_type;
+    size_t num_iter;
+    size_t num_groups;
+    int32_t initial_batch;
+    std::vector<int64_t> qkv_order;
+};
+
+class SDPAWithKVCacheTest : public ::testing::Test, public ::testing::WithParamInterface<Params> {
+public:
+    void test_smoke_multiple_iterations_stateful(const Params& p) {
+        SKIP_IF_CURRENT_TEST_IS_DISABLED();
+#if defined(ANDROID)
+        GTEST_SKIP();
+#endif
+        auto core = ov::test::utils::PluginCache::get().core();
+
+        ov::AnyMap properties = {ov::hint::inference_precision(ov::element::f16),
+                                 ov::intel_gpu::hint::enable_sdpa_optimization(true)};
+
+        const size_t n_heads = 16;
+        const size_t n_features = 64;
+        const size_t context_size = 7;
+
+        const std::vector<int64_t>& qkv_order = p.qkv_order;
+
+        ov::element::Type element_type = p.model_element_type;
+
+        const bool stateful = true;
+        const bool causal = false;
+        const bool with_mask = p.with_mask;
+        const bool with_scale = false;
```

Comment:
Shouldn't all of these values be obtained from test params?

Reply:
Updated and added scale handling in subgraph builder